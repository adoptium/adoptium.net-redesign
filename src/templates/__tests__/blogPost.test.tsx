import React from "react"
import { render } from "@testing-library/react"
import BlogPost, { Head, formatDiv } from "../../templates/blogPost"
import { describe, expect, it, vi } from "vitest"
import { axe } from "vitest-axe"
import { createSingleMDXData } from "../../__fixtures__/page"

vi.mock("../../components/News/RelatedArticles", () => {
  return {
    default: () => <div>related-articles</div>,
  }
})

let mockData = createSingleMDXData()
const pageContext = {
  previous: {
    fields: {
      postPath: "/blog/previous",
    },
    frontmatter: {
      title: "Previous",
    },
  },
  next: {
    fields: {
      postPath: "/blog/next",
    },
    frontmatter: {
      title: "Next",
    },
  },
}

describe("BlogPost Template page", () => {
  it("renders correctly", () => {
    const { container } = render(
      <BlogPost
        data={mockData}
        pageContext={pageContext}
        location={{}}
        children={"sample blog"}
      />,
    )
    // eslint-disable-next-line
    const pageContent = container.querySelector("main")

    expect(pageContent).toMatchSnapshot()
  })

  it("renders correctly - featured image", () => {
    mockData.mdx.frontmatter.featuredImage = {
      childImageSharp: {
        gatsbyImageData: {
          images: {
            fallback: {
              src: "https://fake-image.com",
            },
          },
        },
      },
    }
    const { container } = render(
      <BlogPost
        data={mockData}
        pageContext={pageContext}
        location={{}}
        children={"sample blog"}
      />,
    )
    // eslint-disable-next-line
    const pageContent = container.querySelector("main")

    expect(pageContent).toMatchSnapshot()
  })

  it("formatDiv renders correctly", () => {
    const { container } = render(
      formatDiv({ dangerouslySetInnerHTML: { __html: "<p>test</p>" } }),
    )
    expect(container).toMatchSnapshot()
  })

  it("formatDiv renders correctly - inline code block", () => {
    const { container } = render(
      formatDiv({
        dangerouslySetInnerHTML: {
          __html: '<code class="language-text">test</code>',
        },
      }),
    )
    expect(container).toMatchSnapshot()
  })

  it("head renders correctly", () => {
    const { container } = render(<Head data={mockData} />)
    // eslint-disable-next-line
    const title = container.querySelector("title")
    expect(title?.textContent).toEqual("MDX Page title | Adoptium")
  })

  it("has no accessibility violations", async () => {
    const { container } = render(
      <BlogPost
        data={mockData}
        pageContext={pageContext}
        location={{}}
        children={"sample blog"}
      />,
    )
    const results = await axe(container)
    expect(results).toHaveNoViolations()
  })
})
