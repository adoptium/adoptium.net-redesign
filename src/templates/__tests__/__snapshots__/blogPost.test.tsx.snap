// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BlogPost Template page > formatDiv renders correctly - inline code block 1`] = `
<div>
  <code>
    <code
      class="language-text"
    >
      test
    </code>
  </code>
</div>
`;

exports[`BlogPost Template page > formatDiv renders correctly 1`] = `
<div>
  <div>
    <p>
      test
    </p>
  </div>
</div>
`;

exports[`BlogPost Template page > renders correctly - featured image 1`] = `
<main>
  <div
    class="pt-48 pb-12"
  >
    <div
      class="mx-auto max-w-[832px] w-full px-6 lg:px-0 flex flex-col items-center justify-center"
    >
      <div
        class="self-stretch h-52 flex-col justify-center items-center gap-6 flex"
      >
        <div
          class="self-stretch h-32 flex-col justify-center items-center gap-4 flex"
        >
          <div
            class="justify-start items-center gap-3 inline-flex"
          >
            <svg
              fill="none"
              height="14"
              viewBox="0 0 14 14"
              width="14"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="#FF1464"
                height="8"
                rx="2"
                width="8"
                x="3"
                y="3"
              />
              <rect
                height="11"
                rx="3.5"
                stroke="#FF1464"
                stroke-opacity="0.25"
                stroke-width="3"
                width="11"
                x="1.5"
                y="1.5"
              />
            </svg>
            <div
              class="text-rose-600 text-base font-bold leading-normal"
            >
              News article
            </div>
          </div>
          <h1
            class="self-stretch text-center text-white   text-[36px] lg:text-[48px] leading-[122.222%] lg:leading-[116.667%] font-semibold"
          >
            MDX Page title
          </h1>
        </div>
        <div
          class="self-stretch text-center text-grey text-grey-300 text-xl font-normal leading-7"
        >
          MDX Page excerpt
        </div>
        <div
          class="flex justify-center items-center gap-5"
        >
          <img
            alt="Adoptium PMC"
            image="[object Object]"
            stlye="[object Object]"
            style="margin-left: 0.5rem; margin-bottom: 0px; min-width: 30px; border-radius: 100%;"
          />
          <span
            class="text-[16px] font-bold leading-[150%] text-white"
          >
            <a
              href="/blog/author/pmc"
            >
              <span
                class="text-primary pr-2"
              >
                Adoptium PMC
                ,
              </span>
               
              2021-01-01
            </a>
          </span>
        </div>
      </div>
    </div>
  </div>
  <section
    class="mx-auto w-full p-6 lg:px-0 flex flex-col items-center justify-center"
  >
    <div
      class="max-w-4xl"
    >
      <article>
        <header
          class="pb-5"
        >
          <div
            id="share-buttons"
          >
            <button
              aria-label="Share on X (Twitter)"
              class="react-share__ShareButton mr-2"
              networkname="twitter"
              style="background-color: transparent; padding: 0px; cursor: pointer;"
            >
              <svg
                fill="currentColor"
                height="25"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="25"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M389.2 48h70.6L305.6 224.2 487 464H345L233.7 318.6 106.5 464H35.8L200.7 275.5 26.8 48H172.4L272.9 180.9 389.2 48zM364.4 421.8h39.1L151.1 88h-42L364.4 421.8z"
                />
              </svg>
            </button>
            <button
              aria-label="Share on Linkedin"
              class="react-share__ShareButton mr-2"
              networkname="linkedin"
              style="background-color: transparent; padding: 0px; cursor: pointer;"
            >
              <svg
                fill="currentColor"
                height="25"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 448 512"
                width="25"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"
                />
              </svg>
            </button>
            <button
              aria-label="Share on Facebook"
              class="react-share__ShareButton mr-2"
              networkname="facebook"
              style="background-color: transparent; padding: 0px; cursor: pointer;"
            >
              <svg
                fill="currentColor"
                height="25"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="25"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M504 256C504 119 393 8 256 8S8 119 8 256c0 123.78 90.69 226.38 209.25 245V327.69h-63V256h63v-54.64c0-62.15 37-96.48 93.67-96.48 27.14 0 55.52 4.84 55.52 4.84v61h-31.28c-30.8 0-40.41 19.12-40.41 38.73V256h68.78l-11 71.69h-57.78V501C413.31 482.38 504 379.78 504 256z"
                />
              </svg>
            </button>
            <button
              aria-label="Share on Reddit"
              class="react-share__ShareButton mr-2"
              networkname="reddit"
              style="background-color: transparent; padding: 0px; cursor: pointer;"
            >
              <svg
                fill="currentColor"
                height="25"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="25"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M201.5 305.5c-13.8 0-24.9-11.1-24.9-24.6 0-13.8 11.1-24.9 24.9-24.9 13.6 0 24.6 11.1 24.6 24.9 0 13.6-11.1 24.6-24.6 24.6zM504 256c0 137-111 248-248 248S8 393 8 256 119 8 256 8s248 111 248 248zm-132.3-41.2c-9.4 0-17.7 3.9-23.8 10-22.4-15.5-52.6-25.5-86.1-26.6l17.4-78.3 55.4 12.5c0 13.6 11.1 24.6 24.6 24.6 13.8 0 24.9-11.3 24.9-24.9s-11.1-24.9-24.9-24.9c-9.7 0-18 5.8-22.1 13.8l-61.2-13.6c-3-.8-6.1 1.4-6.9 4.4l-19.1 86.4c-33.2 1.4-63.1 11.3-85.5 26.8-6.1-6.4-14.7-10.2-24.1-10.2-34.9 0-46.3 46.9-14.4 62.8-1.1 5-1.7 10.2-1.7 15.5 0 52.6 59.2 95.2 132 95.2 73.1 0 132.3-42.6 132.3-95.2 0-5.3-.6-10.8-1.9-15.8 31.3-16 19.8-62.5-14.9-62.5zM302.8 331c-18.2 18.2-76.1 17.9-93.6 0-2.2-2.2-6.1-2.2-8.3 0-2.5 2.5-2.5 6.4 0 8.6 22.8 22.8 87.3 22.8 110.2 0 2.5-2.2 2.5-6.1 0-8.6-2.2-2.2-6.1-2.2-8.3 0zm7.7-75c-13.6 0-24.6 11.1-24.6 24.9 0 13.6 11.1 24.6 24.6 24.6 13.8 0 24.9-11.1 24.9-24.6 0-13.8-11-24.9-24.9-24.9z"
                />
              </svg>
            </button>
            <button
              aria-label="Share on Bluesky"
              class="react-share__ShareButton mr-2"
              networkname="bluesky"
              style="background-color: transparent; padding: 0px; cursor: pointer;"
            >
              <svg
                fill="currentColor"
                height="25"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 576 512"
                width="25"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M407.8 294.7c-3.3-.4-6.7-.8-10-1.3c3.4 .4 6.7 .9 10 1.3zM288 227.1C261.9 176.4 190.9 81.9 124.9 35.3C61.6-9.4 37.5-1.7 21.6 5.5C3.3 13.8 0 41.9 0 58.4S9.1 194 15 213.9c19.5 65.7 89.1 87.9 153.2 80.7c3.3-.5 6.6-.9 10-1.4c-3.3 .5-6.6 1-10 1.4C74.3 308.6-9.1 342.8 100.3 464.5C220.6 589.1 265.1 437.8 288 361.1c22.9 76.7 49.2 222.5 185.6 103.4c102.4-103.4 28.1-156-65.8-169.9c-3.3-.4-6.7-.8-10-1.3c3.4 .4 6.7 .9 10 1.3c64.1 7.1 133.6-15.1 153.2-80.7C566.9 194 576 75 576 58.4s-3.3-44.7-21.6-52.9c-15.8-7.1-40-14.9-103.2 29.8C385.1 81.9 314.1 176.4 288 227.1z"
                />
              </svg>
            </button>
            <button
              aria-label="Share via Email"
              class="react-share__ShareButton mr-2"
              networkname="email"
              style="background-color: transparent; padding: 0px; cursor: pointer;"
            >
              <svg
                fill="currentColor"
                height="25"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="25"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M48 64C21.5 64 0 85.5 0 112c0 15.1 7.1 29.3 19.2 38.4L236.8 313.6c11.4 8.5 27 8.5 38.4 0L492.8 150.4c12.1-9.1 19.2-23.3 19.2-38.4c0-26.5-21.5-48-48-48L48 64zM0 176L0 384c0 35.3 28.7 64 64 64l384 0c35.3 0 64-28.7 64-64l0-208L294.4 339.2c-22.8 17.1-54 17.1-76.8 0L0 176z"
                />
              </svg>
            </button>
          </div>
        </header>
        <article
          class="prose prose-invert lg:prose-lg max-w-none"
        >
          sample blog
        </article>
        <div
          class="hidden md:flex justify-between items-center gap-10 py-10"
        >
          <div
            class="bg-[#3E3355] w-full h-[1px] "
          />
          <div
            class="flex flex-col items-center"
          >
            <p
              class="tab-button-text"
            >
              Share this post
            </p>
            <div
              class="flex items-center gap-4"
            >
              <a
                href="#share-buttons"
              >
                <button
                  class="bg-primary px-6 py-3 rounded-[80px] tab-button-text flex items-center gap-3"
                >
                  <svg
                    fill="currentColor"
                    height="25"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 24 24"
                    width="25"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      id="Share_2"
                    >
                      <path
                        d="M18.44,15.94a2.5,2.5,0,0,0-1.96.95L7.97,12.64a2.356,2.356,0,0,0,0-1.29l8.5-4.25a2.5,2.5,0,1,0-.53-1.54,2.269,2.269,0,0,0,.09.65l-8.5,4.25a2.5,2.5,0,1,0,0,3.08l8.5,4.25a2.269,2.269,0,0,0-.09.65,2.5,2.5,0,1,0,2.5-2.5Zm0-11.88a1.5,1.5,0,1,1-1.5,1.5A1.5,1.5,0,0,1,18.44,4.06ZM5.56,13.5A1.5,1.5,0,1,1,7.06,12,1.5,1.5,0,0,1,5.56,13.5Zm12.88,6.44a1.5,1.5,0,1,1,1.5-1.5A1.5,1.5,0,0,1,18.44,19.94Z"
                      />
                    </g>
                  </svg>
                  Share
                </button>
              </a>
            </div>
          </div>
          <div
            class="bg-[#3E3355] w-full h-[1px] inline-block"
          />
        </div>
      </article>
      <div>
        <ul
          class="flex flex-wrap justify-between list-none p-0"
        >
          <li>
            <a
              href="/blog/next"
              rel="next"
            >
              ← 
              Next
            </a>
          </li>
          <li>
            <a
              href="/blog/previous"
              rel="prev"
            >
              Previous
               →
            </a>
          </li>
        </ul>
      </div>
    </div>
  </section>
  <div>
    related-articles
  </div>
</main>
`;

exports[`BlogPost Template page > renders correctly 1`] = `
<main>
  <div
    class="pt-48 pb-12"
  >
    <div
      class="mx-auto max-w-[832px] w-full px-6 lg:px-0 flex flex-col items-center justify-center"
    >
      <div
        class="self-stretch h-52 flex-col justify-center items-center gap-6 flex"
      >
        <div
          class="self-stretch h-32 flex-col justify-center items-center gap-4 flex"
        >
          <div
            class="justify-start items-center gap-3 inline-flex"
          >
            <svg
              fill="none"
              height="14"
              viewBox="0 0 14 14"
              width="14"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="#FF1464"
                height="8"
                rx="2"
                width="8"
                x="3"
                y="3"
              />
              <rect
                height="11"
                rx="3.5"
                stroke="#FF1464"
                stroke-opacity="0.25"
                stroke-width="3"
                width="11"
                x="1.5"
                y="1.5"
              />
            </svg>
            <div
              class="text-rose-600 text-base font-bold leading-normal"
            >
              News article
            </div>
          </div>
          <h1
            class="self-stretch text-center text-white   text-[36px] lg:text-[48px] leading-[122.222%] lg:leading-[116.667%] font-semibold"
          >
            MDX Page title
          </h1>
        </div>
        <div
          class="self-stretch text-center text-grey text-grey-300 text-xl font-normal leading-7"
        >
          MDX Page excerpt
        </div>
        <div
          class="flex justify-center items-center gap-5"
        >
          <img
            alt="Adoptium PMC"
            image="[object Object]"
            stlye="[object Object]"
            style="margin-left: 0.5rem; margin-bottom: 0px; min-width: 30px; border-radius: 100%;"
          />
          <span
            class="text-[16px] font-bold leading-[150%] text-white"
          >
            <a
              href="/blog/author/pmc"
            >
              <span
                class="text-primary pr-2"
              >
                Adoptium PMC
                ,
              </span>
               
              2021-01-01
            </a>
          </span>
        </div>
      </div>
    </div>
  </div>
  <section
    class="mx-auto w-full p-6 lg:px-0 flex flex-col items-center justify-center"
  >
    <div
      class="max-w-4xl"
    >
      <article>
        <header
          class="pb-5"
        >
          <div
            id="share-buttons"
          >
            <button
              aria-label="Share on X (Twitter)"
              class="react-share__ShareButton mr-2"
              networkname="twitter"
              style="background-color: transparent; padding: 0px; cursor: pointer;"
            >
              <svg
                fill="currentColor"
                height="25"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="25"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M389.2 48h70.6L305.6 224.2 487 464H345L233.7 318.6 106.5 464H35.8L200.7 275.5 26.8 48H172.4L272.9 180.9 389.2 48zM364.4 421.8h39.1L151.1 88h-42L364.4 421.8z"
                />
              </svg>
            </button>
            <button
              aria-label="Share on Linkedin"
              class="react-share__ShareButton mr-2"
              networkname="linkedin"
              style="background-color: transparent; padding: 0px; cursor: pointer;"
            >
              <svg
                fill="currentColor"
                height="25"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 448 512"
                width="25"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"
                />
              </svg>
            </button>
            <button
              aria-label="Share on Facebook"
              class="react-share__ShareButton mr-2"
              networkname="facebook"
              style="background-color: transparent; padding: 0px; cursor: pointer;"
            >
              <svg
                fill="currentColor"
                height="25"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="25"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M504 256C504 119 393 8 256 8S8 119 8 256c0 123.78 90.69 226.38 209.25 245V327.69h-63V256h63v-54.64c0-62.15 37-96.48 93.67-96.48 27.14 0 55.52 4.84 55.52 4.84v61h-31.28c-30.8 0-40.41 19.12-40.41 38.73V256h68.78l-11 71.69h-57.78V501C413.31 482.38 504 379.78 504 256z"
                />
              </svg>
            </button>
            <button
              aria-label="Share on Reddit"
              class="react-share__ShareButton mr-2"
              networkname="reddit"
              style="background-color: transparent; padding: 0px; cursor: pointer;"
            >
              <svg
                fill="currentColor"
                height="25"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="25"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M201.5 305.5c-13.8 0-24.9-11.1-24.9-24.6 0-13.8 11.1-24.9 24.9-24.9 13.6 0 24.6 11.1 24.6 24.9 0 13.6-11.1 24.6-24.6 24.6zM504 256c0 137-111 248-248 248S8 393 8 256 119 8 256 8s248 111 248 248zm-132.3-41.2c-9.4 0-17.7 3.9-23.8 10-22.4-15.5-52.6-25.5-86.1-26.6l17.4-78.3 55.4 12.5c0 13.6 11.1 24.6 24.6 24.6 13.8 0 24.9-11.3 24.9-24.9s-11.1-24.9-24.9-24.9c-9.7 0-18 5.8-22.1 13.8l-61.2-13.6c-3-.8-6.1 1.4-6.9 4.4l-19.1 86.4c-33.2 1.4-63.1 11.3-85.5 26.8-6.1-6.4-14.7-10.2-24.1-10.2-34.9 0-46.3 46.9-14.4 62.8-1.1 5-1.7 10.2-1.7 15.5 0 52.6 59.2 95.2 132 95.2 73.1 0 132.3-42.6 132.3-95.2 0-5.3-.6-10.8-1.9-15.8 31.3-16 19.8-62.5-14.9-62.5zM302.8 331c-18.2 18.2-76.1 17.9-93.6 0-2.2-2.2-6.1-2.2-8.3 0-2.5 2.5-2.5 6.4 0 8.6 22.8 22.8 87.3 22.8 110.2 0 2.5-2.2 2.5-6.1 0-8.6-2.2-2.2-6.1-2.2-8.3 0zm7.7-75c-13.6 0-24.6 11.1-24.6 24.9 0 13.6 11.1 24.6 24.6 24.6 13.8 0 24.9-11.1 24.9-24.6 0-13.8-11-24.9-24.9-24.9z"
                />
              </svg>
            </button>
            <button
              aria-label="Share on Bluesky"
              class="react-share__ShareButton mr-2"
              networkname="bluesky"
              style="background-color: transparent; padding: 0px; cursor: pointer;"
            >
              <svg
                fill="currentColor"
                height="25"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 576 512"
                width="25"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M407.8 294.7c-3.3-.4-6.7-.8-10-1.3c3.4 .4 6.7 .9 10 1.3zM288 227.1C261.9 176.4 190.9 81.9 124.9 35.3C61.6-9.4 37.5-1.7 21.6 5.5C3.3 13.8 0 41.9 0 58.4S9.1 194 15 213.9c19.5 65.7 89.1 87.9 153.2 80.7c3.3-.5 6.6-.9 10-1.4c-3.3 .5-6.6 1-10 1.4C74.3 308.6-9.1 342.8 100.3 464.5C220.6 589.1 265.1 437.8 288 361.1c22.9 76.7 49.2 222.5 185.6 103.4c102.4-103.4 28.1-156-65.8-169.9c-3.3-.4-6.7-.8-10-1.3c3.4 .4 6.7 .9 10 1.3c64.1 7.1 133.6-15.1 153.2-80.7C566.9 194 576 75 576 58.4s-3.3-44.7-21.6-52.9c-15.8-7.1-40-14.9-103.2 29.8C385.1 81.9 314.1 176.4 288 227.1z"
                />
              </svg>
            </button>
            <button
              aria-label="Share via Email"
              class="react-share__ShareButton mr-2"
              networkname="email"
              style="background-color: transparent; padding: 0px; cursor: pointer;"
            >
              <svg
                fill="currentColor"
                height="25"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="25"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M48 64C21.5 64 0 85.5 0 112c0 15.1 7.1 29.3 19.2 38.4L236.8 313.6c11.4 8.5 27 8.5 38.4 0L492.8 150.4c12.1-9.1 19.2-23.3 19.2-38.4c0-26.5-21.5-48-48-48L48 64zM0 176L0 384c0 35.3 28.7 64 64 64l384 0c35.3 0 64-28.7 64-64l0-208L294.4 339.2c-22.8 17.1-54 17.1-76.8 0L0 176z"
                />
              </svg>
            </button>
          </div>
        </header>
        <article
          class="prose prose-invert lg:prose-lg max-w-none"
        >
          sample blog
        </article>
        <div
          class="hidden md:flex justify-between items-center gap-10 py-10"
        >
          <div
            class="bg-[#3E3355] w-full h-[1px] "
          />
          <div
            class="flex flex-col items-center"
          >
            <p
              class="tab-button-text"
            >
              Share this post
            </p>
            <div
              class="flex items-center gap-4"
            >
              <a
                href="#share-buttons"
              >
                <button
                  class="bg-primary px-6 py-3 rounded-[80px] tab-button-text flex items-center gap-3"
                >
                  <svg
                    fill="currentColor"
                    height="25"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 24 24"
                    width="25"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      id="Share_2"
                    >
                      <path
                        d="M18.44,15.94a2.5,2.5,0,0,0-1.96.95L7.97,12.64a2.356,2.356,0,0,0,0-1.29l8.5-4.25a2.5,2.5,0,1,0-.53-1.54,2.269,2.269,0,0,0,.09.65l-8.5,4.25a2.5,2.5,0,1,0,0,3.08l8.5,4.25a2.269,2.269,0,0,0-.09.65,2.5,2.5,0,1,0,2.5-2.5Zm0-11.88a1.5,1.5,0,1,1-1.5,1.5A1.5,1.5,0,0,1,18.44,4.06ZM5.56,13.5A1.5,1.5,0,1,1,7.06,12,1.5,1.5,0,0,1,5.56,13.5Zm12.88,6.44a1.5,1.5,0,1,1,1.5-1.5A1.5,1.5,0,0,1,18.44,19.94Z"
                      />
                    </g>
                  </svg>
                  Share
                </button>
              </a>
            </div>
          </div>
          <div
            class="bg-[#3E3355] w-full h-[1px] inline-block"
          />
        </div>
      </article>
      <div>
        <ul
          class="flex flex-wrap justify-between list-none p-0"
        >
          <li>
            <a
              href="/blog/next"
              rel="next"
            >
              ← 
              Next
            </a>
          </li>
          <li>
            <a
              href="/blog/previous"
              rel="prev"
            >
              Previous
               →
            </a>
          </li>
        </ul>
      </div>
    </div>
  </section>
  <div>
    related-articles
  </div>
</main>
`;
